File: public-transport-web-server/src/main/java/pl/wrona/webserver/otp/OtpService.java
package pl.wrona.webserver.otp;

import lombok.AllArgsConstructor;
import org.igeolab.iot.otp.api.model.Route;
import org.igeolab.iot.otp.api.model.RoutesResponse;
import org.igeolab.iot.otp.api.model.Stop;
import org.igeolab.iot.otp.api.model.StopDetails;
import org.igeolab.iot.otp.api.model.StopTime;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.math.BigDecimal;
import java.util.List;

@Service
@AllArgsConstructor
public class OtpService {

    private final OtpClient otpClient;

    public ResponseEntity<RoutesResponse> planTrip(String instanceId, String fromPlace, String toPlace, String date, String time, String mode, String local, Boolean showIntermediateStops, Double maxWalkDistance, Boolean arriveBy, Boolean wheelchair, Integer numItineraries, Boolean realtime, String optimize) {
        ResponseEntity<RoutesResponse> response = this.otpClient.planTrip(instanceId, fromPlace, toPlace, date, time, mode, local, showIntermediateStops, maxWalkDistance, arriveBy, wheelchair, numItineraries, realtime, optimize);
        return ResponseEntity.ok(response.getBody());
    }

    public ResponseEntity<List<Stop>> getStops(String instanceId, BigDecimal maxLat, BigDecimal minLon, BigDecimal minLat, BigDecimal maxLon) {
        ResponseEntity<List<Stop>> response = this.otpClient.getStops(instanceId, maxLat, minLon, minLat, maxLon);
        return ResponseEntity.ok(response.getBody());
    }

    public ResponseEntity<StopDetails> getStopDetails(String instanceId, String stopId) {
        ResponseEntity<StopDetails> response = this.otpClient.getStopDetails(instanceId, stopId);
        return ResponseEntity.ok(response.getBody());
    }

    public ResponseEntity<List<Route>> getRoutes(String instanceId, String stopId) {
        return this.otpClient.getRoutes(instanceId, stopId);
    }

    public ResponseEntity<List<StopTime>> getStoptimes(String instanceId, String stopId, String date) {
        return this.otpClient.getStoptimes(instanceId, stopId, date);
    }

}

File: public-transport-web-client/src/app/http/otp.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from "@angular/common/http";
import {Observable} from "rxjs";
import moment from "moment";
import {RoutesResponse, Stop, StopDetails, Route, StopTime} from "../generated";

@Injectable({
  providedIn: 'root'
})
export class OtpService {

  constructor(private httpClient: HttpClient) {
  }

  plan(startLat: number, startLon: number, endLat: number, endLon: number, departureDate: moment.Moment): Observable<RoutesResponse> {
    return this.httpClient.get<RoutesResponse>(`/api/otp/routers/default/plan?fromPlace=${startLat},${startLon}&toPlace=${endLat},${endLon}&date=${departureDate.format('yyyy-MM-dd')}&time=${departureDate.format('HH:mm')}&mode=TRANSIT&showIntermediateStops=true&local=en`);
  }

  planBike(startLat: number, startLon: number, endLat: number, endLon: number, departureDate: moment.Moment): Observable<RoutesResponse> {
    return this.httpClient.get<RoutesResponse>(`/api/otp/routers/default/plan?fromPlace=${startLat},${startLon}&toPlace=${endLat},${endLon}&date=${departureDate.format('yyyy-MM-dd')}&time=${departureDate.format('HH:mm')}&mode=BICYCLE_RENT&showIntermediateStops=true&local=en`);
  }

  getStops(maxLat: number, minLon: number, minLat: number, maxLon: number): Observable<Stop[]> {
      const headerDict = {
          'Content-Type': 'application/json',
          'Accept': 'application/json'
      }

    return this.httpClient.get<Stop[]>(`/api/otp/routers/default/index/stops?maxLat=${maxLat}&minLon=${minLon}&minLat=${minLat}&maxLon=${maxLon}`, {headers: new HttpHeaders(headerDict)});
  }

  getStopDetails(stopId: string): Observable<StopDetails> {
    return this.httpClient.get<StopDetails>(`/api/otp/routers/default/index/stops/${stopId}`);
  }

  getRoutes(stopId: string): Observable<Route[]> {
    return this.httpClient.get<Route[]>(`/api/otp/routers/default/index/stops/${stopId}/routes`);
  }

  getStopTimes(stopId: string, date: moment.Moment): Observable<StopTime[]> {
    return this.httpClient.get<StopTime[]>(`/api/otp/routers/default/index/stops/${stopId}/stoptimes/${date.format('yyyyMMDD')}`);
  }
}
